### PR TITLE
code review #1

### PR DESCRIPTION
**Pull request by HypeDis**

### CODE REVIEW

File: server/express.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+const chalk = require('chalk');
```
Comment (HypeDis) [inline server/express.js]: swap names with index.js

File: server/index.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    secret: 'hopethisworks',
+    resave: false,
+    cookie: {
+      maxAge: 7.2 * Math.exp(10, 6) // 2 hours
+    }
+  })
+);
+
+// session logging
+app.use((req, res, next) => {
```
Comment (HypeDis) [inline server/index.js]: logging stuff should be locked behind development mode

File: server/index.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    .then(userOrNull => {
+      if (!userOrNull) req.loggedIn = false;
+      else {
+        req.loggedIn = true;
+        req.user = userOrNull;
+      }
+      next();
+    })
+    .catch(e => {
+      console.log('error searching for a user by session.userId');
```
Comment (HypeDis) [inline server/index.js]: if you want to use for debugging log actual userId as well

File: server/index.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+app.use('/auth', require('./auth'));
+
+// static middleware
+app.use(express.static(path.join(__dirname, '../public')));
+
+// got to use those routes!
+app.use('/api', require('./api'));
+
+// handling errors
+app.use((err, req, res, next) => {
```
Comment (HypeDis) [inline server/index.js]: move error middleware to bottom

File: webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+const { resolve } = require('path')
+
+module.exports = {
+  entry: ['babel-polyfill', './app/main'],
+  output: {
+    path: __dirname,
+    filename: './public/bundle.js'
+  },
+  mode: 'development',
```
Comment (HypeDis) [inline webpack.config.js]: dont hardcode mode in the webpack config

File: server/db/index.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+  Products,
+  Categories,
+  Sizes,
+  ShoeSizes,
+  Genders,
+  Colors,
+  ProductListings
+} = require('./product_models/index');
+
+const { Orders, OrderStatuses, OrderItems } = require('./order_models/index');
```
Comment (HypeDis) [inline server/db/index.js]: A little heavy handed, dont need to make everything a reference. perhaps enum?

File: server/db/index.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+//////////////////////////
+// PRODUCT ASSOCIATIONS //
+//////////////////////////
+
+// each product has only one category
+Categories.hasMany(Products);
+Products.belongsTo(Categories);
+// each product has only one size
+Sizes.hasMany(Products);
+Products.belongsTo(Sizes);
```
Comment (HypeDis) [inline server/db/index.js]: a little weird

Comment (HypeDis) [reply server/db/index.js]: are different product size combos unique?

File: server/db/order_models/order.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+const { NOW, DECIMAL, DATE, UUID, UUIDV4 } = Sequelize;
+
+const Orders = db.define('orders', {
+  id: {
+    type: UUID,
+    defaultValue: UUIDV4,
+    primaryKey: true
+  },
+  totalCost: {
+    type: DECIMAL,
```
Comment (HypeDis) [inline server/db/order_models/order.js]: add precision to decimal

File: server/db/order_models/orderItems.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+const { UUID, UUIDV4, INTEGER, DECIMAL } = Sequelize;
+
+const OrderItems = db.define('orderItems', {
+  id: {
+    type: UUID,
+    defaultValue: UUIDV4,
+    primaryKey: true
+  },
+  //NOTE: Added quantity in this model so we can keep track in the database when pulling up past orders in the UI. Added unitPrice into model so we can keep track of price in case we do a promo code that will change the unitPrice. This may be good if we want to display all the individual items in the order when selecting order history. This would also be how we access the different prices / quantity of items in an order vs the Products info from the 'include' Products model relationship which displayed the product's info rather the carts. http://localhost:3000/api/orders/<insertorderId>
+  quantity: {
```
Comment (HypeDis) [inline server/db/order_models/orderItems.js]: think about what would happen if you ordered 3 of the same item and returned one of them

File: server/db/order_models/orderStatus.js [hunk 10]
```diff
@@ -0,0 +1,37 @@
+const Sequelize = require('sequelize');
+const db = require('../database');
```
Comment (HypeDis) [inline server/db/order_models/orderStatus.js]: this table probably not necessary

File: server/db/product_models/category.js [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+  {
+    id: {
+      type: INTEGER,
+      primaryKey: true,
+      validate: {
+        min: 1,
+        max: 8 // just guessing how many categories we want
+      }
+    },
+    category: {
```
Comment (HypeDis) [inline server/db/product_models/category.js]: instead of enum have category_name

Comment (HypeDis) [reply server/db/product_models/category.js]: perhaps a parent category id as well?

File: server/db/product_models/color.js [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+    id: {
+      type: INTEGER,
+      primaryKey: true,
+      validate: {
+        min: 1,
+        max: 9
+      }
+    },
+    color: {
+      type: ENUM,
```
Comment (HypeDis) [inline server/db/product_models/color.js]: don't use enum

File: server/db/product_models/gender.js [hunk 13]
```diff
@@ -0,0 +1,29 @@
+const Sequelize = require('sequelize');
+const db = require('../database');
+
+const { ENUM, INTEGER } = Sequelize;
+
+const Genders = db.define(
```
Comment (HypeDis) [inline server/db/product_models/gender.js]: probably could include a gender enum in the product table and remove this table

File: app/components/root.jsx [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+    store.dispatch(fetchProducts());
+    store.dispatch(fetchOrders());
+    store.dispatch(initialLogInAttempt());
+    // store.dispatch(fetchUsers());
+  }
+  render() {
+    return (
+      <Router>
+        <main>
+          <Route component={Nav} />
```
Comment (HypeDis) [inline app/components/root.jsx]: why use a route component?

File: app/redux/cart.js [hunk 15]
```diff
@@ -0,0 +1,138 @@
...
+export const editQuantity = orderItem => {
+  return {
+    type: EDIT_QUANTITY,
+    orderItem
+  };
+};
+
+// thunks
+
+export const fetchOrder = activeOrder => {
```
Comment (HypeDis) [inline app/redux/cart.js]: reconcilliation of merging guest cart and user cart is bugged

File: server/index.js (hunk 4 above)
Comment (hertzj) [reply server/index.js]: JH

File: server/db/index.js (hunk 6 above)
Comment (hertzj) [reply server/db/index.js]: JL

File: server/db/index.js (hunk 7 above)
Comment (hertzj) [reply server/db/index.js]: JL

File: server/db/order_models/order.js (hunk 8 above)
Comment (hertzj) [reply server/db/order_models/order.js]: JL

File: server/db/order_models/orderItems.js (hunk 9 above)
Comment (hertzj) [reply server/db/order_models/orderItems.js]: JL

File: server/db/order_models/orderStatus.js (hunk 10 above)
Comment (hertzj) [reply server/db/order_models/orderStatus.js]: JL

File: server/db/product_models/category.js (hunk 11 above)
Comment (hertzj) [reply server/db/product_models/category.js]: JL

File: server/db/product_models/color.js (hunk 12 above)
Comment (hertzj) [reply server/db/product_models/color.js]: JL

File: server/db/product_models/gender.js (hunk 13 above)
Comment (hertzj) [reply server/db/product_models/gender.js]: JL

File: app/components/root.jsx (hunk 14 above)
Comment (hertzj) [reply app/components/root.jsx]: JH

File: app/redux/cart.js (hunk 15 above)
Comment (hertzj) [reply app/redux/cart.js]: suggestion: active users will always post to DB and NOT local storage
on sign in, if there are items in local storage:
1) if no active cart, POST active cart and POST associated cartItems from local storage
2) if active cart, POST local storage cartItems as associated to activeCart 
3) guests always use local storage (and redux store) till checkout; AT checkout, we POST a "user" for them with name, email, and address info, POST an associated order, and POST orderItems associated to that order

**Pull request merged**